From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mm: constify page_address_in_vma()
Date: Tue, 23 Jul 2024 16:34:57 +0100

If we also mark the struct folio argument to folio_anon_vma(), we can make
page_address_in_vma() take a const struct page pointer.

Link: https://lkml.kernel.org/r/20240723153503.1669586-3-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/rmap.h |    2 +-
 mm/internal.h        |    2 +-
 mm/rmap.c            |    5 +++--
 mm/util.c            |    2 +-
 4 files changed, 6 insertions(+), 5 deletions(-)

--- a/include/linux/rmap.h~mm-constify-page_address_in_vma
+++ a/include/linux/rmap.h
@@ -732,7 +732,7 @@ bool page_vma_mapped_walk(struct page_vm
 /*
  * Used by swapoff to help locate where page is expected in vma.
  */
-unsigned long page_address_in_vma(struct page *, struct vm_area_struct *);
+unsigned long page_address_in_vma(const struct page *, struct vm_area_struct *);
 
 /*
  * Cleans the PTEs of shared mappings.
--- a/mm/internal.h~mm-constify-page_address_in_vma
+++ a/mm/internal.h
@@ -784,7 +784,7 @@ static inline bool free_area_empty(struc
 }
 
 /* mm/util.c */
-struct anon_vma *folio_anon_vma(struct folio *folio);
+struct anon_vma *folio_anon_vma(const struct folio *folio);
 
 #ifdef CONFIG_MMU
 void unmap_mapping_folio(struct folio *folio);
--- a/mm/rmap.c~mm-constify-page_address_in_vma
+++ a/mm/rmap.c
@@ -771,9 +771,10 @@ static bool should_defer_flush(struct mm
  * At what user virtual address is page expected in vma?
  * Caller should check the page is actually part of the vma.
  */
-unsigned long page_address_in_vma(struct page *page, struct vm_area_struct *vma)
+unsigned long page_address_in_vma(const struct page *page,
+		struct vm_area_struct *vma)
 {
-	struct folio *folio = page_folio(page);
+	const struct folio *folio = page_folio(page);
 	pgoff_t pgoff;
 
 	if (folio_test_anon(folio)) {
--- a/mm/util.c~mm-constify-page_address_in_vma
+++ a/mm/util.c
@@ -809,7 +809,7 @@ void *vcalloc_noprof(size_t n, size_t si
 }
 EXPORT_SYMBOL(vcalloc_noprof);
 
-struct anon_vma *folio_anon_vma(struct folio *folio)
+struct anon_vma *folio_anon_vma(const struct folio *folio)
 {
 	unsigned long mapping = (unsigned long)folio->mapping;
 
_
